.github/workflows/pre-commit.yaml

```
@@ -0,0 +1,72 @@
+        with:
+          ref: ${{ github.ref_name }} # explicitly get the head of the branch, which will include any new commits pushed if this is a dependabot branch
+          persist-credentials: false
+
+      - name: Checkout code not during push
+        if: ${{ github.event_name != 'push' }}
+        uses: actions/checkout@v5.0.0
+        with:
+          persist-credentials: false
+
+      - name: Install latest versions of packages
+        uses: ./.github/actions/install_deps
```

actionlint (1.7.9)

could not parse action metadata in "/home/jailuser/git/.github/actions/install_deps": line 4: unexpected key "type" for definition of input "skip-updating-devcontainer-hash"

(action)